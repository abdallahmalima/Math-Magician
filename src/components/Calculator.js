import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculateObject, setCalculateObject] = useState({
    total: null,
    next: null,
    operation: null,

  });

  const handleClick = (e) => {
    const buttonName = e.target.innerText;
    const obj = calculate(calculateObject, buttonName);
    setCalculateObject(obj);
  };

  return <div className='calculator-wrapper'>
    <div data-testid='screen' className='screen'>
        <p>{calculateObject.total}
        {calculateObject.operation}
        {calculateObject.next}</p>
    </div>
    <div className='btn-container'>
     <div className='btn-row'>

    <button data-testid='AC' className='btn' onClick={ handleClick}>AC</button>

    <button data-testid='+/-' className='btn' onClick={ handleClick}>+/-</button>

    <button data-testid='%' className='btn' onClick={ handleClick}>%</button>

    <button data-testid='รท' className='btn btn-last' onClick={ handleClick}>&divide;</button>

    </div>
    <div className='btn-row'>

    <button data-testid='7' className='btn' onClick={ handleClick}>7</button>

    <button data-testid='8' className='btn' onClick={ handleClick}>8</button>

    <button data-testid='9' className='btn' onClick={ handleClick}>9</button>

    <button data-testid='x' className='btn btn-last' onClick={ handleClick}>x</button>

    </div>
    <div className='btn-row'>

    <button data-testid='4' className='btn' onClick={ handleClick}>4</button>

    <button data-testid='5' className='btn' onClick={ handleClick}>5</button>

    <button data-testid='6' className='btn'onClick={ handleClick}>6</button>

    <button data-testid='-' className='btn btn-last' onClick={ handleClick}>-</button>

    </div>
    <div className='btn-row'>

    <button data-testid='1' className='btn' onClick={ handleClick}>1</button>

    <button data-testid='2' className='btn' onClick={ handleClick}>2</button>

    <button data-testid='3' className='btn' onClick={ handleClick}>3</button>

    <button data-testid='+' className='btn btn-last' onClick={ handleClick}>+</button>

    </div>
    <div className='btn-row'>

    <button data-testid='0' className='btn btn-zero' onClick={ handleClick}>0</button>

    <button data-testid='.' className='btn' onClick={ handleClick}>.</button>

    <button data-testid='=' className='btn btn-last' onClick={ handleClick}>=</button>

    </div>

    </div>
    </div>;
};

export default Calculator;