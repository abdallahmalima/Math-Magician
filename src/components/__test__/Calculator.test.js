import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

it('check all 19 calculator buttons in the the page', () => {
  render(<Calculator />);
  const buttonElements = screen.getAllByRole('button');
  expect(buttonElements.length).toBe(19);
});

it('it add 7 and 3 By click button', () => {
  render(<Calculator />);

  const getFirtBtn = screen.getByTestId('7');
  const getSecondBtn = screen.getByTestId('3');
  const getOpBtn = screen.getByTestId('+');
  const getEqualBtn = screen.getByTestId('=');
  const calScreen = screen.getByTestId('screen');

  fireEvent.click(getFirtBtn, { target: { innerText: '7' } });
  fireEvent.click(getOpBtn, { target: { innerText: '+' } });
  fireEvent.click(getSecondBtn, { target: { innerText: '3' } });
  fireEvent.click(getEqualBtn, { target: { innerText: '=' } });

  expect(calScreen.textContent).toBe('10');
});

it('check component match with snapshot', () => {
 const calComponent= render(<Calculator />);
  expect(calComponent).toMatchSnapshot();
});
