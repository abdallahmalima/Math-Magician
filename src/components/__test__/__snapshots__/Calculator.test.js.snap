// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check component match with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator-wrapper"
      >
        <div
          class="screen"
          data-testid="screen"
        >
          <p />
        </div>
        <div
          class="btn-container"
        >
          <div
            class="btn-row"
          >
            <button
              class="btn"
              data-testid="AC"
            >
              AC
            </button>
            <button
              class="btn"
              data-testid="+/-"
            >
              +/-
            </button>
            <button
              class="btn"
              data-testid="%"
            >
              %
            </button>
            <button
              class="btn btn-last"
              data-testid="รท"
            >
              รท
            </button>
          </div>
          <div
            class="btn-row"
          >
            <button
              class="btn"
              data-testid="7"
            >
              7
            </button>
            <button
              class="btn"
              data-testid="8"
            >
              8
            </button>
            <button
              class="btn"
              data-testid="9"
            >
              9
            </button>
            <button
              class="btn btn-last"
              data-testid="x"
            >
              x
            </button>
          </div>
          <div
            class="btn-row"
          >
            <button
              class="btn"
              data-testid="4"
            >
              4
            </button>
            <button
              class="btn"
              data-testid="5"
            >
              5
            </button>
            <button
              class="btn"
              data-testid="6"
            >
              6
            </button>
            <button
              class="btn btn-last"
              data-testid="-"
            >
              -
            </button>
          </div>
          <div
            class="btn-row"
          >
            <button
              class="btn"
              data-testid="1"
            >
              1
            </button>
            <button
              class="btn"
              data-testid="2"
            >
              2
            </button>
            <button
              class="btn"
              data-testid="3"
            >
              3
            </button>
            <button
              class="btn btn-last"
              data-testid="+"
            >
              +
            </button>
          </div>
          <div
            class="btn-row"
          >
            <button
              class="btn btn-zero"
              data-testid="0"
            >
              0
            </button>
            <button
              class="btn"
              data-testid="."
            >
              .
            </button>
            <button
              class="btn btn-last"
              data-testid="="
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator-wrapper"
    >
      <div
        class="screen"
        data-testid="screen"
      >
        <p />
      </div>
      <div
        class="btn-container"
      >
        <div
          class="btn-row"
        >
          <button
            class="btn"
            data-testid="AC"
          >
            AC
          </button>
          <button
            class="btn"
            data-testid="+/-"
          >
            +/-
          </button>
          <button
            class="btn"
            data-testid="%"
          >
            %
          </button>
          <button
            class="btn btn-last"
            data-testid="รท"
          >
            รท
          </button>
        </div>
        <div
          class="btn-row"
        >
          <button
            class="btn"
            data-testid="7"
          >
            7
          </button>
          <button
            class="btn"
            data-testid="8"
          >
            8
          </button>
          <button
            class="btn"
            data-testid="9"
          >
            9
          </button>
          <button
            class="btn btn-last"
            data-testid="x"
          >
            x
          </button>
        </div>
        <div
          class="btn-row"
        >
          <button
            class="btn"
            data-testid="4"
          >
            4
          </button>
          <button
            class="btn"
            data-testid="5"
          >
            5
          </button>
          <button
            class="btn"
            data-testid="6"
          >
            6
          </button>
          <button
            class="btn btn-last"
            data-testid="-"
          >
            -
          </button>
        </div>
        <div
          class="btn-row"
        >
          <button
            class="btn"
            data-testid="1"
          >
            1
          </button>
          <button
            class="btn"
            data-testid="2"
          >
            2
          </button>
          <button
            class="btn"
            data-testid="3"
          >
            3
          </button>
          <button
            class="btn btn-last"
            data-testid="+"
          >
            +
          </button>
        </div>
        <div
          class="btn-row"
        >
          <button
            class="btn btn-zero"
            data-testid="0"
          >
            0
          </button>
          <button
            class="btn"
            data-testid="."
          >
            .
          </button>
          <button
            class="btn btn-last"
            data-testid="="
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
